lib/fs.js

```diff
@@ -423,6 +423,9 @@ function tryCreateBuffer(size, fd, isUserFd) {
   var threw = true;
   var buffer;
   try {
+    if (size > kMaxLength) {
+      throw new ERR_FS_FILE_TOO_LARGE(size);
+    }
     buffer = Buffer.allocUnsafe(size);
```

Comment:
Since we now check for `kMaxLength`: is it still possible for the alloc to throw an error? If not, then the `try / catch` should be obsolete.

Reply:
I think it can still throw if there is not enough memory available to construct the buffer?

Reply:
Ah, yes, I thought it might crash in that case but it will actually throw `Array buffer allocation failed`.